components/brave_sync/brave_sync_service_impl.cc

```diff
@@ -480,6 +483,7 @@ void BraveSyncServiceImpl::OnResolvedPreferences(const RecordsList& records) {
           actually_merged);
       contains_only_one_device = sync_devices->size() < 2 &&
         record->action == jslib::SyncRecord::Action::A_DELETE;
+      have_two_or_more_after_merge = actually_merged && sync_devices->size() >= 2;
```

Comment:
we can put `s/had_less_than_two/waiting_for_second_device` here and no need to check > 2 case because StartLoop should only be initialized once.
Should look like this
`second_device_connected = waiting_for_second_device  && actually_merged && sync_devices->size() == 2`

Reply:
thanks, simplified this.